x-pack/plugins/apm/public/components/fleet_integration/apm_policy_form/settings/settings_form.tsx

```diff
@@ -0,0 +1,163 @@
+      const value = vars?.[key]?.value;
+      const { isValid, message } = validateSettingValue(setting, value);
+      return (
+        <React.Fragment key={key}>
+          <EuiDescribedFormGroup
+            title={<h3>{setting.rowTitle}</h3>}
+            description={setting.rowDescription}
+          >
+            <EuiFormRow
+              label={setting.label}
+              isInvalid={!isValid}
+              error={isValid ? undefined : message}
+              helpText={<EuiText size="xs">{setting.helpText}</EuiText>}
+              labelAppend={
+                <EuiText size="xs" color="subdued">
+                  {setting.labelAppend}
+                </EuiText>
+              }
+            >
+              <FormRowSetting
+                setting={setting}
+                onChange={onChange}
+                value={value}
+              />
+            </EuiFormRow>
+          </EuiDescribedFormGroup>
+          {setting.settings &&
+            value &&
+            setting.settings.map((childSettings) =>
+              getSettingFormRow(childSettings)
+            )}
+        </React.Fragment>
+      );
+    }
+  }
+  return getSettingFormRow(initialSetting);
+}
+interface Props {
+  title: string;
+  subtitle: string;
+  settings: SettingDefinition[];
+  vars?: PackagePolicyVars;
+  onChange: FormRowOnChange;
+}
+
+export function SettingsForm({
+  title,
+  subtitle,
+  settings,
+  vars,
+  onChange,
+}: Props) {
+  return (
+    <EuiPanel>
+      <EuiFlexGroup direction="column" gutterSize="s">
+        <EuiFlexItem>
+          <EuiTitle size="xs">
+            <h4>{title}</h4>
+          </EuiTitle>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiText size="s" color="subdued">
+            {subtitle}
+          </EuiText>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiHorizontalRule margin="s" />
+
+      {settings.map((setting) => {
+        return FormRow({
+          initialSetting: setting,
+          vars,
+          onChange,
+        });
+      })}
+    </EuiPanel>
+  );
+}
+
+function AdvancedOptions({ children }: { children: React.ReactNode }) {
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <>
+      <EuiFlexGroup>
+        <EuiFlexItem />
+        <EuiFlexItem>
+          <EuiLink
+            onClick={() => {
+              setIsOpen((state) => !state);
+            }}
+          >
+            <EuiFlexGroup gutterSize="xs" alignItems="center">
+              <EuiFlexItem grow={false}>
+                <EuiIcon type={isOpen ? 'arrowDown' : 'arrowRight'} />
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiText>
+                  {i18n.translate(
+                    'xpack.apm.fleet_integration.settings.advancedOptionsLavel',
+                    { defaultMessage: 'Advanced options' }
+                  )}
+                </EuiText>
+              </EuiFlexItem>
+            </EuiFlexGroup>
```

Comment:
Any reason not just to use an `euiButton` here as in the other advanced options settings button in the main integration configuration?

Reply:
not really, thanks for that!!